src/screens/Profile/PaymentOptions/index.tsx

```diff
@@ -0,0 +1,205 @@
+import SvgDelete from '@atb/assets/svg/mono-icons/actions/Delete';
+import {useAuthState} from '@atb/auth';
+import FullScreenHeader from '@atb/components/screen-header/full-header';
+import ThemeText from '@atb/components/text';
+import PaymentBrand from '@atb/screens/Ticketing/Purchase/Payment/PaymentBrand';
+import {
+  getExpireDate,
+  getPaymentTypeName,
+} from '@atb/screens/Ticketing/Purchase/utils';
+import {StyleSheet, Theme, useTheme} from '@atb/theme';
+import {
+  deleteRecurringPayment,
+  listRecurringPayments,
+  RecurringPayment,
+} from '@atb/tickets';
+import {useTranslation} from '@atb/translations';
+import PaymentOptionsTexts from '@atb/translations/screens/subscreens/PaymentOptions';
+import React, {useEffect, useState} from 'react';
+import {ActivityIndicator, TouchableOpacity, View} from 'react-native';
+import {ScrollView} from 'react-native-gesture-handler';
+import {destructiveAlert} from '../Home/utils';
+import MessageBox from '@atb/components/message-box';
+import {StackNavigationProp} from '@react-navigation/stack';
+import {RootStackParamList} from '@atb/navigation';
+import useFontScale from '@atb/utils/use-font-scale';
+
+type PaymentOptionsProps = {
+  navigation: StackNavigationProp<RootStackParamList>;
+};
+
+export default function PaymentOptions({navigation}: PaymentOptionsProps) {
+  const style = useStyle();
+  const {theme} = useTheme();
+  const {t} = useTranslation();
+  const {user} = useAuthState();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [storedCards, setStoredCards] = useState<RecurringPayment[]>([]);
+
+  async function getRecurringPaymentOptions(): Promise<
+    Array<RecurringPayment>
+  > {
+    if (!user || user.isAnonymous) return [];
+    const recurringOptions: Array<RecurringPayment> =
+      await listRecurringPayments();
+    return [...recurringOptions.reverse()];
+  }
+
+  /*
+   * Refresh stored cards when screen gets focus
+   */
+  useEffect(() => {
+    const unsubscribe = navigation.addListener('focus', refreshCards);
+    return unsubscribe;
+  }, [navigation]);
+
+  async function handleRemovePayment(paymentOption: RecurringPayment) {
+    setIsLoading(true);
+    await deleteRecurringPayment(paymentOption.id);
+    refreshCards();
+  }
+
+  async function refreshCards() {
+    setIsLoading(true);
+    const remoteOptions = await getRecurringPaymentOptions();
```

Comment:
What happens here if this requests fails? I usually do a `try {setLoading(true) } catch (e) {handleError(e)} finally { setLoading(false) }` pattern to ensure that you always make sure to unset loading.

Reply:
yeah good point.